lib/publiccloud/utils.pm

```diff
@@ -49,6 +50,50 @@ sub select_host_console {
     set_var('TUNNELED', $tunneled) if $tunneled;
 }
 
+sub register_addon {
+    my ($remote, $addon) = @_;
+    my $arch = get_var('PUBLIC_CLOUD_ARCH') // "x86_64";
+    $arch = "aarch64" if ($arch eq "arm64");
+    record_info($addon, "Going to register '$addon' addon");
+    if ($addon =~ /ltss/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch, "-r " . get_required_var('SCC_REGCODE_LTSS'));
+    } elsif (is_sle('<15') && $addon =~ /tcm|wsm|contm|asmm|pcm/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '`echo ${VERSION} | cut -d- -f1`', $arch);
+    } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch);
+    } else {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), undef, $arch);
+    }
+}
+
+sub registercloudguest {
+    my ($my_instance, $remote) = @_;
```

Comment:
I would try to make these things more reusable, so `$remote` or `$my_instance` not both.

Maybe only `$remote` and provide a function in `publiccloud::instance` which give this as return value.

Reply:
and make `publiccloud::instance::run_ssh_command` be able to work with `$remote` as well. like
```perl
sub run_ssh_command {
my $self = shift;
if (ref($self) eq 'publiccloud::instance') {
    $args{ssh_opts} //= $self->ssh_opts() . " -i '" . $self->ssh_key . "'";
    $args{remote} //= $self->username() . '@' . $args{public_ip};
   ....
} else  {
  $args{remote} = $self;
}
$self = undef; # just to not confuse the later code...
```

Reply:
but be aware of cylce dependencies...